eclair-core/src/test/scala/fr/acinq/eclair/db/SqlitePaymentsDbSpec.scala

```diff
@@ -158,26 +158,30 @@ class SqlitePaymentsDbSpec extends FunSuite {
 
     val bob = Bob.keyManager
 
-    val (paymentHash1, paymentHash2) = (randomBytes32, randomBytes32)
+    val (paymentHash1, paymentHash2, paymentHash3) = (randomBytes32, randomBytes32, randomBytes32)
 
-    val i1 = PaymentRequest(chainHash = Block.TestnetGenesisBlock.hash, amount = Some(MilliSatoshi(123)), paymentHash = paymentHash1, privateKey = bob.nodeKey.privateKey, description = "Some invoice", expirySeconds = None, timestamp = someTimestamp)
-    val i2 = PaymentRequest(chainHash = Block.TestnetGenesisBlock.hash, amount = None, paymentHash = paymentHash2, privateKey = bob.nodeKey.privateKey, description = "Some invoice", expirySeconds = Some(123456), timestamp = Platform.currentTime.milliseconds.toSeconds)
+    val i1 = PaymentRequest(chainHash = Block.TestnetGenesisBlock.hash, amount = Some(MilliSatoshi(123)), paymentHash = paymentHash1, privateKey = bob.nodeKey.privateKey, description = "Some invoice1", expirySeconds = None, timestamp = someTimestamp)
+    val i2 = PaymentRequest(chainHash = Block.TestnetGenesisBlock.hash, amount = None, paymentHash = paymentHash2, privateKey = bob.nodeKey.privateKey, description = "Some invoice2", expirySeconds = Some(123456), timestamp = Platform.currentTime.milliseconds.toSeconds)
+    val i3 = PaymentRequest(chainHash = Block.TestnetGenesisBlock.hash, amount = Some(MilliSatoshi(123)), paymentHash = paymentHash3, privateKey = bob.nodeKey.privateKey, description = "Some invoice3", expirySeconds = Some(0), timestamp = someTimestamp - 500)
 
-    // i2 doesn't expire
+    // i1 doesn't expire
     assert(i1.expiry.isEmpty && i2.expiry.isDefined)
     assert(i1.amount.isDefined && i2.amount.isEmpty)
 
     db.addPaymentRequest(i1, ByteVector32.Zeroes)
     db.addPaymentRequest(i2, ByteVector32.One)
+    db.addPaymentRequest(i3, ByteVector32.One)
 
     // order matters, i2 has a more recent timestamp than i1
-    assert(db.listPaymentRequests(0, (Platform.currentTime.milliseconds + 1.minute).toSeconds) == Seq(i2, i1))
+    assert(db.listPaymentRequests(0, (Platform.currentTime.milliseconds + 1.minute).toSeconds) == Seq(i2, i1, i3))
     assert(db.getPaymentRequest(i1.paymentHash) == Some(i1))
     assert(db.getPaymentRequest(i2.paymentHash) == Some(i2))
+    assert(db.getPaymentRequest(i3.paymentHash) == Some(i3))
 
     assert(db.listPendingPaymentRequests(0, (Platform.currentTime.milliseconds + 1.minute).toSeconds) == Seq(i2, i1))
     assert(db.getPendingPaymentRequestAndPreimage(paymentHash1) == Some((ByteVector32.Zeroes, i1)))
     assert(db.getPendingPaymentRequestAndPreimage(paymentHash2) == Some((ByteVector32.One, i2)))
+    assert(db.getPendingPaymentRequestAndPreimage(paymentHash3) == None) // expired invoice won't be returned
```

Comment:
This test isn't really meaningful because `getPendingPaymentRequestAndPreimage` uses `Platform.currentTime`, and here we use a constant `someTimestamp` way in the past.

Reply:
In cae244e the test has been improved using a more recent timestamp for the expired invoice.